src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Extensions/MetadataAttributePass.cs

```diff
@@ -19,7 +18,7 @@ internal class MetadataAttributePass : IntermediateNodePassBase, IRazorOptimizat
 
     protected override void OnInitialized()
     {
-        _identifierFeature = Engine.GetFeature<IMetadataIdentifierFeature>();
+        Engine.TryGetFeature(out _identifierFeature);
```

Comment:
Do we want to assert that this is true here?

Reply:
Sure.